Source/App.js

```diff
@@ -1 +1,417 @@
+        }
+    }
+
+    tileStyle.addEventListener('change', set3DTileStyle);
+
+    //////////////////////////////////////////////////////////////////////////
+    // 3D Models and time-dynamic properties
+    //////////////////////////////////////////////////////////////////////////
+
+
+    // Generate a random circular pattern with varying heights.
+    // This will be replaced with CZML file once it's made.
+    var start = viewer.clock.startTime;
+    function computeCircularFlight(lon, lat, radius) {
+        var lonRadians = Cesium.Math.toRadians(lon);
+        var latRadians = Cesium.Math.toRadians(lat);
+        var property = new Cesium.SampledPositionProperty();
+        for (var i = 0; i <= 360; i += 45) {
+            var offset = Cesium.Math.toRadians(i);
+            var time = Cesium.JulianDate.addSeconds(start, i * 240, new Cesium.JulianDate());
+            var position = Cesium.Cartesian3.fromRadians(lonRadians + (radius * 1.5 * Math.cos(offset)), latRadians + (radius * Math.sin(offset)), 500);
+            property.addSample(time, position);
+            //Also create a point for each sample we generate.
+            viewer.entities.add({
+                position : position,
+                point : {
+                    pixelSize : 8,
+                    color : Cesium.Color.YELLOW
+                }
+            });
+        }
+        return property;
+    }
+
+    // Load drone 3D model
+    // Set a starting position in an interesting environment.
+    var position = Cesium.Cartesian3.fromDegrees(-74.01881302800247, 40.72694833660694, 700);
+    var heading = Cesium.Math.toRadians(135);
+    var pitch = 0;
+    var roll = 0;
+    var hpr = new Cesium.HeadingPitchRoll(heading, pitch, roll);
+    var orientation = Cesium.Transforms.headingPitchRollQuaternion(position, hpr);
+
+    position = computeCircularFlight(-73.95881302800247, 40.78694833660694, 0.0005)
+    var drone = viewer.entities.add({
+        //Set the entity availability to the same interval as the simulation time.
+        availability : new Cesium.TimeIntervalCollection([new Cesium.TimeInterval({
+            start : viewer.clock.startTime,
+            stop : viewer.clock.stopTime
+        })]),
+        name : 'drone',
+        position : position,
+        orientation : new Cesium.VelocityOrientationProperty(position),
+        model : {
+            uri : './Source/SampleData/Models/CesiumDrone.gltf',
+            minimumPixelSize : 128,
+            maximumScale : 2000
+        },
+        //Show the path as a line sampled in 100 second increments.
+        path : {
+            resolution : 100,
+            material : new Cesium.PolylineGlowMaterialProperty({
+                glowPower : 0.1,
+                color : Cesium.Color.YELLOW
+            }),
+            width : 10
+        }
+    });
+
+    // Interpolate smoothly between position sample points.
+    drone.position.setInterpolationOptions({
```

Comment:
computeCircularFlight is a placeholder until I've created a CZML file to read drone positions from.